src/common/mfu_daos.c

```diff
@@ -2607,21 +2607,22 @@ static int serialize_recx_array(struct hdf5_args *hdf5,
     int                 attr_num = 0;
     int                 buf_len = 0;
     int                 path_len = 0;
-    int                 encode_buf_len;
+    int                 encode_buf_len = 0;
     uint32_t            number = 5;
-    size_t              nalloc;
+    size_t              nalloc = 0;
     daos_anchor_t       recx_anchor = {0}; 
     daos_anchor_t       fetch_anchor = {0}; 
-    daos_epoch_range_t  eprs[5];
-    daos_recx_t         recxs[5];
+    daos_epoch_range_t  eprs[5] = {0};
+    daos_recx_t         recxs[5] = {0};
     daos_size_t         size = 0;
-    char                attr_name[ATTR_NAME_LEN];
-    char                number_str[ATTR_NAME_LEN];
-    char                attr_num_str[ATTR_NAME_LEN];
+    char                attr_name[ATTR_NAME_LEN] = {0};
+    char                number_str[ATTR_NAME_LEN] = {0};
+    char                attr_num_str[ATTR_NAME_LEN] = {0};
     unsigned char       *encode_buf = NULL;
-    d_sg_list_t         sgl;
-    d_iov_t             iov;
+    d_sg_list_t         sgl = {0};
+    d_iov_t             iov = {0};
```

Comment:
@daltonbohning I was getting some errors from valgrind related to using stack allocations before initialization, but all of them weren't necessary to fix the errors. I don't think it hurts anything and it can remove some of the memsets.